### ISSUE TITLE
postgresql_database idempotency

### ISSUE THREAD

**Issue by Plork** [sous-chefs/postgresql#514]
### Cookbook version
6.1.1 from 'master' commit: d476cdf

### Chef-client version
13.8.5

### Platform Details
Centos-7.3 (bento)

### Scenario:
Creating a Database/ and User as owner.

#### attributes.rb
```ruby
default['testdatabase']['db_name'] = 'database'
default['testdatabase']['db_username'] = 'user'
default['testdatabase']['db_password'] = 'password'
```

#### recipe.rb
```ruby
postgresql_server_install 'PostgreSQL Package' do
  version '9.6'
  action %i(install create)
end

postgresql_user 'PostgreSQL User' do
  action :create
  user node['testdatabase']['db_username']
  password node['testdatabase']['db_password']
end

postgresql_database 'PostgreSQL Database' do
  database node['testdatabase']['db_name']
  action :create
  owner node['testdatabase']['db_username']
end
```

### Actual Result:
Running the recipe the second time gives
` STDERR: createdb: database creation failed: ERROR:  database "database" already exists`

The first time the recipe runs fine. Seems the "database_exists?" doesn't work.
Running the command manually gives:

```
-bash-4.2$ psql -c "SELECT datname from pg_database WHERE datname='database'" postgres | grep database
 database
```

**Comment by Plork** [comment]
Adding some logging in the `:create` action gives:
```
* log[false] action write
* bash[Create Database database] action run
```

```ruby
  log database_exists?(new_resource)

  bash "Create Database #{new_resource.database}" do
    code createdb
    user new_resource.user
    not_if { database_exists?(new_resource) }
  end
```

**Comment by damacus** [comment]
Fixed by #515
